### PR TITLE
enhance: Add `MemoryHighSyncPolicy` back to write buffer manager

### PR DESCRIPTION
**Pull request by congqixia**
See also #27675

This PR adds back MemoryHighSyncPolicy implementation. Also change MinSegmentSize & CheckInterval to configurable param item.

### CODE REVIEW

File: pkg/util/paramtable/component_param.go [hunk 1]
```diff
@@ -2939,6 +2942,31 @@ func (p *dataNodeConfig) init(base *BaseTable) {
...
+	p.MemoryForceSyncSegmentMinNum = ParamItem{
+		Key:          "datanode.memory.forceSyncSegmentMinNum",
+		Version:      "2.4.0",
+		DefaultValue: "1",
+	}
+	p.MemoryForceSyncSegmentMinNum.Init(base.mgr)
+
+	p.MemoryForceSyncMinSize = ParamItem{
+		Key:          "datanode.memory.forceSyncMinSize",
+		Version:      "2.3.6",
```
Comment (XuanYang-cn) [inline pkg/util/paramtable/component_param.go:2954]: What's the difference between 2.3.6 and 2.4.0 for MemoryForceSyncSegmentMinNum

File: internal/datanode/writebuffer/write_buffer.go [hunk 2]
```diff
@@ -162,6 +166,40 @@ func (wb *writeBufferBase) GetFlushTimestamp() uint64 {
...
+		size += segBuf.MemorySize()
+	}
+	return size
+}
+
+func (wb *writeBufferBase) EvictBuffer(policies ...SyncPolicy) {
+	wb.mut.Lock()
+	defer wb.mut.Unlock()
+
+	log := log.Ctx(context.Background()).With(
```
Comment (XuanYang-cn) [inline internal/datanode/writebuffer/write_buffer.go:184]: why using background log? should the syncSegments ctx be the same with this ctx?

**Review by XuanYang-cn — CHANGES_REQUESTED** [review]

File: pkg/util/paramtable/component_param.go (hunk 1 above)
Comment (congqixia) [reply pkg/util/paramtable/component_param.go:2954]: planed to add this param item in 2.3.6, removing it

File: internal/datanode/writebuffer/write_buffer.go (hunk 2 above)
Comment (congqixia) [reply internal/datanode/writebuffer/write_buffer.go:184]: EvictBuffer & SealSegment are different API. EvictBuffer does not have context here

**Pull request merged**